csrc/fusion_segmenter.cpp

```diff
@@ -26,8 +26,129 @@ namespace {
 
 using GroupSet = VectorOfUniqueEntries<SegmentedGroup*>;
 
+// This helper function converts pointer values into a one-to-one integer id.
+// It is used during serialization to define the container's state.
+template <typename ContainerT, typename T>
+std::vector<int64_t> convertPointerToInteger(
```

Comment:
This helper doesn't leverage the fact that the key is a pointer, so I think we can make this more general with almost the same amount of code. 

```
template <typename K, typename V, typename ContainerK>
std::vector<V> convertContainer(const ContainerK& container, const std::unordered_map<K, V>& map) {
```